Orbit.h

```diff
@@ -9,45 +9,173 @@
+//   ORBIT_STOP();
+// }
+//
+// Parameters:
+// name: [const char*] Label to be displayed on current time slice (kMaxEventStringSize characters).
+// col: [orbit::Color] User defined color for the current time slice.
+//
 #define ORBIT_START(name) ORBIT_START_WITH_COLOR(name, orbit::Color::kAuto)
 #define ORBIT_START_WITH_COLOR(name, col) orbit_api::Start(name, col)
 #define ORBIT_STOP() orbit_api::Stop()
 
-// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: profile time span across threads.
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: Profile time spans across scopes or threads.
+//
+// Overview:
+// Async time spans can be started in one scope and stopped on another. They will be displayed
+// in Orbit on a track uniquely identified by the "name" parameter. Note that those time slices
+// do not represent hierarchical information.
+//
+// Note:
+// We limit the maximum number of characters of the "name" parameter to "kMaxEventStringSize". This
+// limitation will be lifted as we roll out a new dynamic instrumentation implementation on Linux.
+// It is possible however to add per-time slice additional information using the ASYNC_STRING macro.
+//
+// Example usage: Tracking "File IO" operations.
+// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);  // File IO request site.
+// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
+// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);  // File IO result site.
+//
+// Parameters:
+// name: [const char*] Name of the *track* that will display the async events in Orbit.
+// id: [uint64_t] A user-provided unique id for the time slice.
+// col: [orbit::Color] User defined color for the current time slice.
+//
 #define ORBIT_START_ASYNC(name, id) ORBIT_START_ASYNC_WITH_COLOR(name, id, orbit::Color::kAuto)
 #define ORBIT_START_ASYNC_WITH_COLOR(name, id, col) orbit_api::StartAsync(name, id, col)
 #define ORBIT_STOP_ASYNC(id) orbit_api::StopAsync(id)
+
+// ORBIT_ASYNC_STRING: Provide additional information for an async time span.
+//
+// Overview:
+// Provide additional information to be displayed on the time slice corresponding to "id".
+//
+// Example usage: Tracking "File IO" operations.
+// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);
+// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
+// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);
+//
+// Parameters:
+// str: [const char*] Arbitrary length string to display in the time slice corresponding to "id".
+// id: [uint64_t] A user-provided unique id for the time slice.
+// col: [orbit::Color] User defined color for the current string.
+//
 #define ORBIT_ASYNC_STRING(str, id) orbit_api::AsyncString(str, id, orbit::Color::kAuto)
 #define ORBIT_ASYNC_STRING_WITH_COLOR(str, id, col) orbit_api::AsyncString(str, id, col)
 
-// ORBIT_[type]: graph variables.
+// ORBIT_[type]: Graph variables.
+//
+// Overview:
+// Send values to be plotted in a track uniquely identified by "name".
```

Comment:
Mention that these are plotted over time? It says this below, but it's easier to understand if you state it here already.